wakepy/methods/wsl.py

```diff
@@ -0,0 +1,100 @@
+import enum
+from abc import ABC, abstractmethod
+from subprocess import Popen, PIPE
+from typing import List
+
+from wakepy.core import DbusAdapter, Method, ModeName, PlatformName
+
+class Flags(enum.StrEnum):
+    KEEP_RUNNING = "$ES_CONTINUOUS -bor $ES_SYSTEM_REQUIRED"
+    KEEP_PRESENTING = "$ES_CONTINUOUS -bor $ES_SYSTEM_REQUIRED -bor $ES_DISPLAY_REQUIRED"
+    RELEASE = "$ES_CONTINUOUS"
+
+
+class WslSetThreadExecutionState(Method, ABC):
+    """This is a method which calls the SetThreadExecutionState function from
+    the kernel32.dll using PowerShell.exe. The SetThreadExecutionState informs the system that it
+    is in use preventing the system from entering sleep or turning off
+    the display while the application is running" (depending on the used
+    flags)."""
+
+    # This should support Windows 7 and above out of the box.  Windows XP and above if PowerShell 2 is installed.
+    supported_platforms = (PlatformName.WSL,)
+
+    def __init__(self, dbus_adapter: DbusAdapter | None = None):
+        self._powershell: Popen = None
+
+        super().__init__(dbus_adapter)
+
+    def enter_mode(self):
+        self._powershell = Popen(["powershell.exe"], stdout=PIPE, stderr=None, stdin=PIPE, universal_newlines=True)
+
+        # C# Code to allow PowerShell to call SetThreadExecutionState
+        code = """$code = @'
```

Comment:
I wonder if we could use textwrap.dedent to make this indented, too?

We could also include the explanation for the non-obvious SetLastError. Perhaps

```
SetLastError makes it possible to look up the result of the function (currently not used, though).
```